### PR TITLE
Searchable models

### PR DESCRIPTION
**Pull request by patriceac**
Makes the models dropdowns searchable (folder or file) and adds a reload models button.
Also makes it easy to add new model dropdowns - e.g. for the model merger - and takes care of refreshing these seamlessly upon reloading.

Something tells me this is going to be a long code review. 😇

image

### CODE REVIEW

File: ui/media/js/searchable-models.js [hunk 1]
```diff
@@ -0,0 +1,586 @@
...
+            if (sibling.style.display == 'list-item') return sibling
+            lastSibling = sibling
+            sibling = sibling.nextElementSibling
+        }
+        
+        // no more siblings, look for next parent if any
+        if (lastSibling.nextElementSibling) {
+            return this.findNextSibling(lastSibling.nextElementSibling.firstElementChild.firstElementChild, false)
+        }
+        else if (lastSibling.parentElement.parentElement.nextElementSibling && lastSibling.parentElement.parentElement.nextElementSibling.firstElementChild && lastSibling.parentElement.parentElement.nextElementSibling.firstElementChild.firstElementChild) {
```
Comment (cmdr2) [inline ui/media/js/searchable-models.js:144]: Any way we can avoid this? :)

**Comment by cmdr2** [comment]
Thanks, I'm okay with this PR getting merged, except that the sibling-related function looks really strange and fragile :) I expect you also feel that way (regarding it being a bit fragile/odd), so can we figure out a better way to do the sibling-check?

Otherwise, this one looks good to merge. Thanks

**Comment by patriceac** [comment]
I hated writing this clause tbh. While it's reliable, it's NOT elegant or
easy to read by any stretch of the imagination. Let me think about it more
and see if there's a better way.

File: ui/media/js/searchable-models.js (hunk 1 above)
Comment (patriceac) [reply ui/media/js/searchable-models.js]: Rewrote the findSibling() functions for improved readability and with proper comments. Hope you like it. :)

**Comment by patriceac** [comment]
@cmdr2 findSibling() functions updated and conflicts resolved. Please let me know what you think of the updated functions.

**Comment by cmdr2** [comment]
Thanks! :)

**Pull request merged**